src/views/user/EditProfilePage.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link, withRouter } from "react-router-dom"
+import validator from 'validator';
+import PropTypes from 'prop-types';
+import updateProfile from '../../actions/updateProfile';
+import EditProfile from '../../components/EditProfile.jsx';
+
+class EditProfilePage extends Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.mouseOutHandler = this.mouseOutHandler.bind(this);
+    this.handleUpdateProfile = this.handleUpdateProfile.bind(this);
+    this.state = {};
+  }
+
+
+
+  handleChange(event) {
+    event.persist();
+    const { name, value, id } = event.target;
+    if((name === 'firstName' || name === 'lastName' || name === 'email'|| name === 'bio')
+    && (value.length === 0 || value.length >= 2)) {
+      let eleClass = event.target.className;
+      if(eleClass.indexOf('field-length-fail') >= 0) {
+        eleClass = eleClass.replace('field-length-fail', 'field-length-ok');
+        const ele = document.getElementById(id);
+        ele.setAttribute('class', eleClass);
+        const spanEle = event.target.nextSibling;
+        spanEle.innerHTML = '';
+      }
+    }
+    if (name === 'email' && (validator.isEmail(value))) {
+      let eleClass = event.target.className;
+      if(eleClass.indexOf('invalid-email') >= 0) {
+        eleClass = eleClass.replace('invalid-email', 'valid-email');
+        const ele = document.getElementById(id);
+        ele.setAttribute('class', eleClass);
+        const spanEle = event.target.nextSibling;
+        spanEle.innerHTML = '';
+       return;
+      }
+    }
+    this.setState(state => (
+      {
+        formData: {
+          ...state.formData, [name]: value,
+        }
+      }
+    ));
+  };
+
+  componentWillMount() {
```

Comment:
Why don't you use `componentDidMount`, because `componentWillMount` is deprecated

Reply:
Ok!